### PR TITLE
Added pre-push hook to ensure tests are run

### PR DESCRIPTION
**Pull request by n1889**
hook will be applied on initial run of run_tests.sh

### CODE REVIEW

**Comment by jimrollenhagen** [comment]
So, two things here:
- This is super nice for people that want to do this. However, it doesn't work for e.g. shared folders with a VM (coding on host, testing on guest), and there's probably people that don't want to run tests on push anyway (at least myself). I don't generally like the idea of forcing this on people.
- I really don't like run_tests.sh installing git hooks. Imagine someone sneaks in a malicious `scripts/githooks/postpush` or whatever. It should be caught in review, but nobody is perfect. 

What if, instead, we updated the testing doc in `doc/dev/` and added this there. "Here's a git hook if you'd like to run tests before pushing each time" etc.

**Comment by ghost** [comment]
@jimrollenhagen +1 On all this. I often push code as WIP to my branches, especially if I'm hopping between machines, and not all tests may be passing. 

Leave it up to CI to run the tests officially. The coders can run the tests locally however they see fit.

**Comment by ghost** [comment]
@n1889 Any resolutions on this?

**Comment by thomwiggers** [comment]
Yeah complete +1 on what @jimrollenhagen said. I don't want pushes prevented, all non-trivial (aka non-documentation typo) pushes to master should go via pull request and pull requests are be checked by CI.

Meanwhile I don't like being prevented to push changes to WIP branches where we might not have working unit tests yet: thon't won't get merged anyway.

Moving this to documentation would be nice though.

**Comment by n1889** [comment]
I'm more than happy to move them to docs if needed, I'm used to using pre-push test hooks so it's second nature to me, to use them. I'll update the PR when I get a chance to.

**Comment by ghost** [comment]
Thanks @n1889

**Comment by ghost** [comment]
@n1889 Just a bump on the changes for this to docs.

**Comment by ghost** [comment]
The other alternative here is to add a ./scripts/add_test_push_hook and ./scripts/remove_test_push_hook

**Comment by ghost** [comment]
@n1889 What's the status of this?

**Comment by n1889** [comment]
Sorry for the delay, been absolutely swamped with work.

File: README.md [hunk 1]
```diff
@@ -54,3 +54,7 @@ Running tests
     ./run_tests.sh -t
     # optionally, just lint
     ./run_tests.sh -l
+
+    # install git hook to automatically run test suite before pushing to remote
```
Comment (thomwiggers) [inline README.md]: I'd prepend `(optional)` to this.

**Comment by thomwiggers** [comment]
Looks good, except for the changes to `.meteor/version` which I can't see any reason for.

**Comment by ghost** [comment]
Should the script to install the hooks be in doc/dev? It seems out of place to have a script in a doc/ directory.

**Comment by ghost** [comment]
@guitarman0831 @n1889 Should be installed under scripts/ in path that makes sense.

**Comment by ghost** [comment]
@n1889 How's this coming along?

**Comment by ghost** [comment]
I'm taking this over.

**Pull request closed without merging**